includes/mailpoet/class-sensei-mailpoet.php

```diff
@@ -0,0 +1,374 @@
+		}
+		return $this->mail_poet_lists;
+	}
+
+	/**
+	 * Add students as subscribers to lists on MailPoet for courses/groups.
+	 *
+	 * @since $$next-version$$
+	 * @param array      $subscribers A list of students belonging to this group/course.
+	 * @param string|int $list_id ID of the list on MailPoet.
+	 *
+	 * @return void
+	 */
+	public function add_subscribers( $subscribers, $list_id ) {
+		$options = array(
+			'send_confirmation_email'      => false,
+			'schedule_welcome_email'       => false,
+			'skip_subscriber_notification' => true,
+		);
+		foreach ( $subscribers as $subscriber ) {
+			$subscriber_data = array(
+				'email'      => $subscriber->user_email,
+				'first_name' => $subscriber->display_name,
+			);
+
+			try {
+				// All WordPress users are already on a list 'WordPress Users' on MailPoet.
+				$mp_subscriber = $this->mailpoet_api->getSubscriber( $subscriber->user_email );
+				$this->mailpoet_api->subscribeToList( $mp_subscriber['id'], $list_id, $options );
+			} catch ( \MailPoet\API\MP\v1\APIException $exception ) {
+				if ( 4 === $exception->getCode() ) {
+					// subscriber does not exist.
+					$this->mailpoet_api->addSubscriber( $subscriber_data, array( $list_id ), $options );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Remove students as subscribers from lists on MailPoet for courses/groups.
+	 *
+	 * @since $$next-version$$
+	 * @param array      $subscribers A list of students to remove.
+	 * @param string|int $list_id ID of the list on MailPoet.
+	 *
+	 * @return void
+	 */
+	public function remove_subscribers( $subscribers, $list_id ) {
+		foreach ( $subscribers as $subscriber ) {
+			try {
+				$mp_subscriber = $this->mailpoet_api->getSubscriber( $subscriber->user_email );
+				$this->mailpoet_api->unsubscribeFromList( $mp_subscriber['id'], $list_id );
+			} catch ( \MailPoet\API\MP\v1\APIException $exception ) {
+				continue;
+			}
+		}
+	}
+
+	/**
+	 * Creates a Sensei LMS MailPoet list with name and description.
+	 *
+	 * @since $$next-version$$
+	 * @param string $list_name The name of the list.
+	 * @param string $list_description The description of the list.
+	 *
+	 * @return int|string|null
+	 */
+	public function create_list( $list_name, $list_description ) {
+		$new_list = array(
+			'name'        => $list_name,
+			'description' => $list_description,
+		);
+		try {
+			$new_list = $this->mailpoet_api->addList( $new_list );
+			return $new_list['id'];
+		} catch ( \MailPoet\API\MP\v1\APIException $exception ) {
+			// see https://github.com/mailpoet/mailpoet/blob/trunk/doc/api_methods/AddList.md#error-handling.
+			return null;
+		}
+	}
+
+	/**
+	 * Add a student as a subscriber to a list on MailPoet for a course or group.
+	 * If the list does not exist, it will be created.
+	 *
+	 * @param int $post_id Post ID.
+	 * @param int $user_id User ID.
+	 *
+	 * @access private
+	 * @return void
+	 */
+	public function add_student_subscriber( $post_id, $user_id ) {
```

Comment:
With the refactor to `sync_subscribers`, this and `remove_student_subscribers` can be removed, right?

Reply:
@yscik  These methods are used by the hooks that add/remove students from courses/groups. Are you suggesting we remove the hooks? 
If yes I'm not against that since the sync logic should take care of scheduling adding/removing students from lists.

Reply:
Oh I've mixed up the methods, sorry, nothing to do for these.